packages/stream-caip10-link-handler/src/caip10-link-handler.ts

```diff
@@ -110,11 +110,26 @@ export class Caip10LinkHandler implements StreamHandler<Caip10Link> {
     }
 
     // TODO: handle CAIP-10 addresses in proof generation of 3id-blockchain-utils
-    const account = validProof.account || validProof.address
-    let [address, chainId] = account.split('@') // eslint-disable-line prefer-const
+    const account: string = validProof.account || validProof.address
 
-    const addressCaip10 = [address, chainId].join('@')
-    if (addressCaip10.toLowerCase() !== state.metadata.controllers[0].toLowerCase()) {
+    let legacyAccountCaip10 = account
+
+    if (!account.includes('@')) {
+      const _accountSplit = account.split(':')
+      const address = _accountSplit.pop()
+      const chainId = _accountSplit.join(':')
+      legacyAccountCaip10 = `${address}@${chainId}`
+    }
+
+    let legacyControllerCaip10 = state.metadata.controllers[0]
+    if (!state.metadata.controllers[0].includes('@')) {
```

Comment:
could this ever happen?  If I'm reading things right, in the Caip10Link class we still always use the legacy format for the controller, right?

Reply:
should this instead just throw an error if the controller is in the legacy format, which we don't expect to ever happen?

Reply:
Yeah, this should never happen.